featuretools/feature_discovery/LiteFeature.py

```diff
@@ -0,0 +1,295 @@
+
+        if self._primitive:
+            assert isinstance(self._primitive, PrimitiveBase)
+            assert (
+                len(self.base_features) > 0
+            ), "there must be base features if given a primitive"
+            self._n_output_features = self._primitive.number_output_features
+            self._depth = max([x.depth for x in self.base_features]) + 1
+            self._alias = self._name
+            self._name = self._primitive.generate_name(
+                [x.name for x in self.base_features],
+            )
+
+        elif self._name is None:
+            raise TypeError("Name must be given if origin feature")
+        elif self._logical_type is None:
+            raise TypeError("Logical Type must be given if origin feature")
+
+        if self._logical_type is not None and "index" not in self._tags:
+            self._tags = self._tags | self._logical_type.standard_tags
+
+        self._id = self._generate_hash()
+
+    @property
+    def name(self):
+        if self._alias:
+            return self._alias
+        elif self.is_multioutput():
+            return f"{self._name}[{self.idx}]"
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._alias = value
+
+    @property
+    def non_indexed_name(self):
+        if not self.is_multioutput():
+            raise ValueError("only used on multioutput features")
+        return self._name
+
+    @property
+    def logical_type(self):
+        return self._logical_type
+
+    @logical_type.setter
+    def logical_type(self, _):
+        raise AttributeError("logical_type is immutable")
+
+    @property
+    def tags(self):
+        return self._tags
+
+    @tags.setter
+    def tags(self, _):
+        raise AttributeError("tags is immutable")
+
+    @property
+    def primitive(self):
+        return self._primitive
+
+    @primitive.setter
+    def primitive(self, _):
+        raise AttributeError("primitive is immutable")
+
+    @property
+    def base_features(self):
+        return self._base_features
+
+    @base_features.setter
+    def base_features(self, _):
+        raise AttributeError("primitive is immutable")
+
+    @property
+    def df_id(self):
+        return self._df_id
+
+    @df_id.setter
+    def df_id(self, _):
+        raise AttributeError("df_id is immutable")
+
+    @property
+    def id(self):
+        return self._id
+
+    @id.setter
+    def id(self, _):
+        raise AttributeError("id is immutable")
+
+    @property
+    def n_output_features(self):
+        return self._n_output_features
+
+    @n_output_features.setter
+    def n_output_features(self, _):
+        raise AttributeError("n_output_features is immutable")
+
+    @property
+    def depth(self):
+        return self._depth
+
+    @depth.setter
+    def depth(self, _):
+        raise AttributeError("depth is immutable")
+
+    @property
+    def related_features(self):
+        return self._related_features
```

Comment:
Similar to `tags` this may also need a `self._related_features.copy()` to avoid mutations being passed to the LiteFeature